sentry_sdk/scope.py

```diff
@@ -208,9 +208,6 @@ def __init__(self, ty=None, client=None):
         incoming_trace_information = self._load_trace_data_from_env()
         self.generate_propagation_context(incoming_data=incoming_trace_information)
 
-        # self._last_event_id is only applicable to isolation scopes
-        self._last_event_id = None  # type: Optional[str]
-
```

Comment:
While I agree that it likely makes sense to move these lines to the `clear` function, I believe this change should be considered in a separate PR, since it does not appear to be needed to fix #3113; the tests you added pass even if I undo this change.